avid-adventurers/src/pages/survey/Rate.tsx

```diff
@@ -1,15 +1,27 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { useNavigate } from 'react-router-dom';
 import { Label } from '../../components/Label/Label';
 import Rating from '../../components/Rating/Rating';
 import { Container, FormWrapper, ButtonRow, BackButton, ContinueButton } from '../styles';
 import { useSurvey } from '../../utils/surveyContext';
 import { LabeledInput } from '../../components/LabeledInput/LabeledInput';
+import ProgressBar from '../../components/ProgressBar/ProgressBar';
 
 
 const Rate: React.FC = () => {
     const navigate = useNavigate();
-    const { rating, setRating, experience, setExperience } = useSurvey();
+    const { rating, setRating } = useSurvey();
+    const [experience, setExperience] = useState('');
+    const pageKey = 'rateExperience';
+
+    useEffect(() => {
+      const savedExperience = localStorage.getItem(pageKey);
+      if (savedExperience) setExperience(savedExperience);
+    }, []);
+
+    useEffect(() => {
+      if (experience) localStorage.setItem(pageKey, experience);
```

Comment:
redundant! you already defined a global experience in surveyContext, so use that